### ISSUE TITLE
I've lost 20+ sessions, Is there a max limit for Saved Sessions? 

### ISSUE THREAD

**Issue by arpanett** [greatsuspender/thegreatsuspender#1158]
Hi I use Brave with Great Suspender, and have recently saved alot of sessions (2-3 per day) for the last 3 weeks. Only today I've noticed that it's stopped showing my new saved sessions, and seemed to have stopped (or maxed out?) at around 22 sessions. Unfortunately I only noticed today and realised that it stopped saving my new sessions around a week ago.

I've tried to delete some sessions in hopes that the newest sessions will show at bottom of list but no luck so far. 

Are these sessions now lost? I tried to search to see if there was some limit set but couldn't find anything. I hadn't updated Brave recently and have had no issues previously with Great Suspender so this is new to me. Thanks.

(Using Brave 1.9.76, Win 8.1 and The Great Suspender
7.1.6)

**UPDATE**: I've noticed that the sessions show up with their names in gsSavedSessions, using the indexeddb in debug view. I wonder if there is a way to easily copy tabs out this way, but it's all drop down menus and quite confusing. But either way the tabs exist, just not within the Great Suspender main extension menu. :cry:

**Comment by deanoemcke** [comment]
There is no limit to the number of saved sessions you can have.
It seems odd that they exist in the indexedDb database but are not showing in the session management screen. Perhaps some sessions contain urls that cause the page to error?

It seems you're competent enough to reach the chrome debug console for the extension, so you may be able to recover your tabs with this script:

```
gsIndexedDb.fetchSavedSessions().then(sessions => {
  sessions.forEach(session => {
    console.log('-------- session: ' + session.name + ' ---------');
    session.windows.forEach(window => {
      console.log('-------- window ---------');
      window.tabs.forEach(tab => {
        console.log(tab.url);
      });
    });
  });
});
```

It's not ideal i know, but it might get you up and running again.

**Comment by arpanett** [comment]
Thank you, that code has saved me! It managed to export all the tabs hidden inside the indexeddb successfully. This saved the 15 sessions that were 'hidden', which contained around 1500 tabs in total. 

I did notice that on the 23rd May is when it stopped showing newly saved sessions in the extension window. And when this happened I was on Brave version v1.8.96. 

I had also seen in my lost sessions was a '**Automatic save point for v7.1.0**' which happened on 29th May, days after sessions had already stopped saving -  so I assume an update was not the cause of this issue. I've also had no crashes recently, or have done any Windows updates that could of caused it to mess up TGS or Brave browser.

I also attempted removing the indexeddb, restarting Brave, closing, and pasting back the indexeddb, but it never showed those hidden sessions again. Grabbing the tabs via indexeddb seems to be the safest fix.

Thank you again for your great work on TGS!
